File: frontend/types/graph.ts
export enum NodeType {
  CONCEPT = 'concept',
  CLAIM = 'claim',
  ASSUMPTION = 'assumption',
  QUESTION = 'question',
  EVIDENCE = 'evidence',
  GAP = 'gap',
  CONTRADICTION = 'contradiction'
}

export enum EdgeType {
  SUPPORTS = 'supports',
  CONTRADICTS = 'contradicts',
  DEPENDS_ON = 'depends_on',
  EXAMPLE_OF = 'example_of'
}

export interface Node {
  id: string;
  type: NodeType;
  text: string;
  confidence: number;
  timestamp: number;
}

export interface Edge {
  id: string;
  source: string;
  target: string;
  type: EdgeType;
  confidence: number;
}

export interface Graph {
  nodes: Node[];
  edges: Edge[];
}

export interface Insight {
  id: string;
  type: 'gap' | 'assumption' | 'contradiction' | 'question';
  title: string;
  description: string;
  relatedNodes: string[];
  timestamp: number;
}

export interface TranscriptSegment {
  id: string;
  text: string;
  timestamp: number;
  speaker?: string;
  nodeIds?: string[];
}

export interface Session {
  id: string;
  title: string;
  createdAt: number;
  updatedAt: number;
  graph: Graph;
  transcript: TranscriptSegment[];
  insights: Insight[];
}

File: backend/app/models/graph.py
from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field


class NodeType(str, Enum):
    CONCEPT = "concept"
    CLAIM = "claim"
    ASSUMPTION = "assumption"
    QUESTION = "question"
    EVIDENCE = "evidence"
    GAP = "gap"
    CONTRADICTION = "contradiction"


class EdgeType(str, Enum):
    SUPPORTS = "supports"
    CONTRADICTS = "contradicts"
    DEPENDS_ON = "depends_on"
    EXAMPLE_OF = "example_of"


class NodeBase(BaseModel):
    node_type: NodeType
    text: str
    confidence: float = Field(ge=0.0, le=1.0)


class NodeCreate(NodeBase):
    pass


class Node(NodeBase):
    node_id: str
    timestamp: int
    session_id: str

    class Config:
        from_attributes = True


class EdgeBase(BaseModel):
    source_node: str
    target_node: str
    relation_type: EdgeType
    confidence: float = Field(ge=0.0, le=1.0)


class EdgeCreate(EdgeBase):
    pass


class Edge(EdgeBase):
    edge_id: str
    session_id: str
    timestamp: int

    class Config:
        from_attributes = True
